app/src/main/java/com/example/b07group7project/create_order/CheckoutFragment.java

```diff
@@ -0,0 +1,86 @@
+package com.example.b07group7project.create_order;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.ImageButton;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.b07group7project.R;
+
+
+import java.util.ArrayList;
+
+public class CheckoutFragment extends Fragment {
+    public CheckoutFragment() {
+        // Empty Constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.checkout_fragment, container, false);
+
+        // Recyler View Stuff
+        RecyclerView recyclerView = view.findViewById(R.id.Checkout_RecyclerView);
+        // TODO: Update Accordinlgy based on Implemenetation of GetCartInterface
+        GetCartInterface cartInterface = new GetCartImplementation();
+        ArrayList<Product> cart = (ArrayList<Product>) cartInterface.getCart("ExampleUser");

```

Comment:
If your casting to ArrayList<Product> anyways, force the getCart() interface to return an arraylist. I mentioned generalizing to List if your implementation doesn't care what kind of list you get. Based on the adapter, you should prefer ArrayList